src/ducks/context/BanksContext.jsx

```diff
@@ -0,0 +1,69 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import React, { createContext, useContext, useEffect, useState } from 'react'
+import PropTypes from 'prop-types'
+import { Q } from 'cozy-client'
+import { KONNECTOR_DOCTYPE } from 'doctypes'
+import { useJobsContext } from 'ducks/context/JobsContext'
+
+export const BanksContext = createContext({})
+
+export const useBanksContext = () => {
+  return useContext(BanksContext)
+}
+
+/**
+ * BanksProvider
+ *
+ * @param children
+ * @param client
+ * @returns {JSX.Element}
+ * @constructor
+ */
+const BanksProvider = ({ children, client }) => {
+  const { jobsInProgress = [] } = useJobsContext()
+  const [banksJobsInProgress, setBanksJobsInProgress] = useState([])
+
+  useEffect(() => {
+    const arr = [...jobsInProgress]
+    if (arr.length > 0) {
+      const formatJobsInProgress = async () => {
+        const ids = arr.map(jobsInProgress => {
+          const slug = jobsInProgress.konnector
+          return `${KONNECTOR_DOCTYPE}/${slug}`
+        })
+        const resp = await client.query(Q(KONNECTOR_DOCTYPE).getByIds(ids))
+        const newJobInProgress = resp.data.map(konnector => {
+          const slug = konnector.slug
+          const jobInProgress = jobsInProgress.find(j => j.konnector === slug)
+          return {
+            konnector: slug,
+            account: jobInProgress.account,
+            institutionLabel: konnector.name
+          }
+        })
+        return newJobInProgress
+      }
+
+      formatJobsInProgress().then(formatJobs => {
+        setBanksJobsInProgress(formatJobs)
+      })
+    }
+  }, [jobsInProgress])
+
+  return (
+    <BanksContext.Provider
+      value={{
+        jobsInProgress: banksJobsInProgress,
+        hasJobsInProgress: banksJobsInProgress.length > 0
```

Comment:
I think the `hasJobsInProgress` is not really adding anything over `banksJobsInProgress.length > 0`. I think it would be best to remove the property as it is a sort of duplicte truth, if we forgot to update `hasJobsInProgress` when changing jobsInProgress, we have a problem.

Reply:
I don't think it necessary to delete it even if the added value is slim. 

I do not see in which case we could have a problem, it is also the responsibility of the developer to make changes when necessary.

Reply:
AMHA the responsibility of the developer is to design data structures that cannot be in a bad state. Here I have the impression that we explicitly ask for a problem. It would be a different thing if hasJobInProgress was a getter that could not be false.

Reply:
I do not really understand where here it could be in bad state ? What is the difference between checking in one place if there are items than doing so in several places? Do you have an example where this could be a problem?

Reply:
1. Here we have a structure that contains twice the same information, there is the length of the array and the array itself. I am not noticing a bug in this code, I am just noticing that duplicating the sources of truth leads to bugs.

2. As an API, the additional field makes it easier to have several users of the code that do things differently. Some will use array.length, and some will use the already computed length. I would find it easier to maintain if it was not possible to use the API in two different ways.

Reply:
ok I see what I mean, thank you for this remark